src/main/java/frc/robot/subsystems/SubsystemCondition.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.subsystems;
+
+import java.util.EnumSet;
+
+public class SubsystemCondition {
+    
+    /**
+     * A list of the possible options conditions can have
+     */
+    public enum ConditionOptions {
+        ENABLED_FOR_TELEOP,
+        ENABLED_FOR_AUTO;
+        
+        // Enum sets of the combinations for the options
+        public static EnumSet<ConditionOptions> TELEOP = EnumSet.of(ENABLED_FOR_TELEOP);
+        public static EnumSet<ConditionOptions> AUTO = EnumSet.of(ENABLED_FOR_AUTO);
+        public static EnumSet<ConditionOptions> BOTH = EnumSet.of(ENABLED_FOR_TELEOP, ENABLED_FOR_AUTO);

```

Comment:
BOTH is requires me to have more context into what this means. How about something more explicit like `TELEP_AND_AUTO`?